quicklens/src/main/scala/com/softwaremill/quicklens/package.scala

```diff
@@ -115,12 +115,12 @@ package object quicklens {
     def each(fa: F[K, T])(f: T => T): F[K, T]
   }
 
-  implicit def mapQuicklensFunctor[K, T] = new QuicklensMapAtFunctor[Map, K, T] {
-    override def at(fa: Map[K, T], key: K)(f: T => T) = {
-      fa.updated(key, f(fa(key)))
+  implicit def mapQuicklensFunctor[M[KT, TT] <: Map[KT, TT], K, T] = new QuicklensMapAtFunctor[M, K, T] {
+    override def at(fa: M[K, T], key: K)(f: T => T) = {
+      fa.updated(key, f(fa(key))).asInstanceOf[M[K, T]]
```

Comment:
If you have an immutable collection you are building a new map anyway. If you have a mutable collection, then you update the map in-place. `updated` takes an implict CanBuildFrom, so I think it's a matter of providing the right instance for the call